c++/src/kj/async-inl.h

```diff
@@ -2247,6 +2295,9 @@ class Coroutine final: public CoroutineBase,
   // a type-erased `await_suspend(stdcoro::coroutine_handle<void>)` override, and implement
   // suspension and resumption in terms of .then(). Yuck!
 
+  template <typename U>
+  Awaiter<U> await_transform(BranchForCoAwaitPromise<U>&& promise) { return Awaiter<U>(promise); }
```

Comment:
We might need an lvalue reference overload, too, unless we know we'll never store a branch as an lvalue before awaiting it.

Edit: I see that this is actually by design, but I'm not sure it's necessary -- see my comment below `addBranchForCoAwait()`.